python/tests/job_runner/test_job_dispatch.py

```diff
@@ -66,36 +68,31 @@ def test_terminate_jobs(self):
             "ert_pid": "",
         }
 
-        job_dispatch_script = os.path.realpath(
-            os.path.join(
-                os.path.dirname(os.path.abspath(__file__)),
-                "../../../bin/job_dispatch.py",
-            )
-        )
-
         with open("jobs.json", "w") as f:
             f.write(json.dumps(self.job_list))
 
-            # Required to execute job_dispatch in separate process group by
-            # os.setsid moves the current process out of the current group
-            with open("job_dispatch_executer", "w") as f:
-                f.write(
-                    "#!/usr/bin/env python\n"
-                    "import os, sys\n"
-                    "os.setsid()\n"
-                    "os.execv(sys.argv[1], sys.argv[1:])\n"
-                    "\n"
+        # macOS doesn't provide /usr/bin/setsid, so we roll our own
+        with open("setsid", "w") as f:
+            f.write(
+                dedent(
+                    """\
+                #!/usr/bin/env python
+                import os
+                import sys
+                os.setsid()
+                os.execvp(sys.argv[1], sys.argv[1:])
+                """
                 )
-            os.chmod(
-                "job_dispatch_executer", stat.S_IRWXU | stat.S_IRWXO | stat.S_IRWXG
             )
+        os.chmod("setsid", 0o755)
 
-        current_dir = os.path.realpath(os.curdir)
+        job_dispatch_script = importlib.util.find_spec("job_runner.job_dispatch").origin
         job_dispatch_process = Popen(
             [
-                os.path.realpath("job_dispatch_executer"),
+                os.getcwd() + "/setsid",
+                sys.executable,
```

Comment:
Why is `sys.executable` added here?

Reply:
The original creates a program specifically for this test. However, `/usr/bin/setsid` exists on Linux, which `exec`s its arguments. But then it turned out that macOS didn't have this utility, so I wrote one in Python. Mostly just shuffling code around tbh.

The reason I'm using `sys.executable` is because I have removed the executable bit from `job_dispatch.py` and removed the shebang, so I have to specify that it should be run with the current Python executable.

Imo, it's a change for the better. Previously it was `#!/usr/bin/env python`, but now we're being explicit about which Python binary we should use.

Reply:
But `job_dispatch.py` is called directly in the wild (for example when submitting to lsf), so we need to shebang, or no? :thinking:

Reply:
https://github.com/equinor/libres/blob/05e06df2926c77f7d9031e3268178afe360f0127/setup.py#L63-L67

Reply:
Basically instead of doing it the incorrect hacky way I decided to do it the Pythonic correct way and let `setuptools`/`pip` handle creation of shebangs.

Reply:
Ahh, I didn't see that! Agree this is much nicer!